TRX/Tron-overview.md

```diff
@@ -183,25 +183,29 @@ try {
+private byte[] generateBlockId(long blockNum, byte[] blockHash) {
+ byte[] numBytes = Longs.toByteArray(blockNum);
+ byte[] hash = blockHash;
+ System.arraycopy(numBytes, 0, hash, 0, 8);
  return hash;
- }
+}
 ```
 BlockHash is the hash of the raw data of the blockheader, which can be calculated as the following:
 ```
 Sha256Hash.of(this.block.getBlockHeader().getRawData().toByteArray())
 ```
 # 10. Construction and signature of transaction
-There are two ways to construct a transaction:
+Based on your own needs, there are two ways to construct the transaction: invoking API on the full node or filling in every fields of a transaction manually at local.
+
 ## 10.1 Invoke APIs on the full node
-Based on your own needs, construct a corresponding local Contract and construct transactions with corresponding APIs. For the contract, please refer to https://github.com/TRONprotocol/protocol/blob/master/core/Contract.proto.
+You could construct transactions with corresponding APIs.
+
+- gRPC API: https://github.com/tronprotocol/Documentation/blob/master/English_Documentation/TRON_Protocol/TRON_Wallet_RPC-API.md
+- HTTP API: https://github.com/tronprotocol/Documentation/blob/master/TRX/Tron-http.md
 
 ## 10.2 Local construction
-Based on the definition of a transaction, you will need to fill in all fields of a transaction to construct a transaction at your local. Please note that you will need to configure the details of reference block and expiration, so you will need to connect to the mainnet during transaction construction. We advise that you set the latest block on the full node as your reference block and production time of the latest block+N minutes as your expiration time. N could be any number you find fit. The backstage condition is (Expiration > production time of the latest block and Expiration < production time of the latest block + 24 hours). If the condition is fulfilled, then the transaction is legit, and if not, the transaction is expired and will not be received by the mainnet.
-method of setting refference block: set RefBlockHash as subarray of newest block's hash from 8 to 16, set BlockBytes as subarray of newest block's height from 6 to 8. The code is as follows:  
+Based on the definition of a transaction, you will need to fill in all fields of a transaction to construct a transaction at your local. Please note that you will need to configure the details of reference block and expiration, so you will need to connect to the mainnet during transaction construction. We advise that you set the latest block on the full node as your reference block and production time of the latest block+N minutes as your expiration time. N could be any number you find fit. The backstage condition is `Expiration > production time of the latest block and Expiration < production time of the latest block + 24 hours`. If the condition is fulfilled, then the transaction is legit, and if not, the transaction is expired and will not be received by the mainnet.
```

Comment:
"Based on the definition of a transaction, you will need to fill in all fields of a transaction to construct a transaction at your local" > "Based on your method of constructing a transaction you are required to populate all fields of a transaction to construct it locally. "

"If the condition is fulfilled, then the transaction is legit, and if not, the transaction is expired and will not be received by the mainnet." > "If the condition is fulfilled, then the transaction is legitimate, and if not, the transaction is expired and will not be acknowledged by the network."